File: DATA_Structure_c++/Linear_search.cpp
#include<iostream>
using namespace std;

void Linear_search(int list[] ,int n, int data){
    int flag = 0;
    for(int i=0; i<n; i++){
        if(list[i] == data){
            cout<<"yes, data is found at index : "<<i<<endl;
            flag = 1;
            break;
         }
    }
    if(flag == 0){
        cout<<"No, data is not found "<<endl;
    }
}

int main(){
int n =6;
int list[n] = {33,22,55,17,39,88};
int data;
cout<<"please enter the data are you saerching for :";
cin>>data;
cout<<"is there is "<<data<<" in the array : ";
Linear_search(list,n,data);



    return 0;
}

File: Data_Structure_inJava/Linear_search.java
package Data_Structure_inJava;

import java.util.Scanner;

public class Linear_search {
   public static void linearSearch(int list[],int n,int data){
     int flag = 0;
     for(int i=0; i<n; i++){
        if(list[i] == data){
            System.out.println("Yes, data is found at index : "+i);
            flag = 1;
            break;
        }
     }
     if(flag == 0){
        System.out.println("No , data is not found");
     }
    }
    public static void main(String[] args) {
        int n = 6;
        int[] list = {20,14,44,53,27,92};
        int data;
        Scanner numb =new Scanner(System.in);
        System.out.print("Enter data you are looking for :");
        data = numb.nextInt();
        System.out.print("is "+data+" found ? :");
        linearSearch(list,n,data);
    }    
}
